### PR TITLE
Filter the output of gitCO

### PR DESCRIPTION
**Pull request by franrivasortega**
I was looking for a way to use `less` within the R console, `page()` output is kinda similar. I still need to finish the last point, but I need to sleep first. I think that creating the pull request at this point is a good idea.

#### To do

- [x] Prevent overloading the console with output when the number of commits in log is high
- [x] Allow the user to specify a date range
- [x] Allow the user to specify **multiple** file names

### CODE REVIEW

**Comment by matloff** [comment]
Excellent, will look and try tomorrow.

Thanks for mentioning neomutt, hadn't been aware of it.

Get some sleep.

Norm

File: R/gitR.R [hunk 1]
```diff
@@ -74,13 +74,22 @@ editPush <- function(fname,commitComment,quiet=FALSE,acceptEnter=FALSE) {
...
+
+   if(is.null(since)) {
+     cmd <- 'git log'
+   } else {
+     cmd <- paste0('git log --since=', since)
+   }
+
+   glog <- system(cmd, intern=TRUE)
+
+   page(trimws(glog), method = "print")
```
Comment (franrivasortega) [inline R/gitR.R]: `trimws()` removes indentation of  the commit message, making it more readable (in my opinion)

File: R/gitR.R [hunk 2]
```diff
@@ -75,18 +75,16 @@ editPush <- function(fname,commitComment,quiet=FALSE,acceptEnter=FALSE) {
...
 
-   if(is.null(since)) {
-     cmd <- 'git log'
-   } else {
-     cmd <- paste0('git log --since=', since)
-   }
+   opt_file <- if(!is.null(file)) paste("--follow", file)
+   opt_since <- if(!is.null(since)) paste0("--since=", since)
 
+   cmd <- paste('git log', opt_since, opt_file)
```
Comment (franrivasortega) [inline R/gitR.R:88]: A better way of building the git command. This way including options is easier.

Comment (franrivasortega) [reply R/gitR.R:88]: As a consequence opt_file needs to be always last in the vector.

**Comment by matloff** [comment]
Everything is very nice, except the 'file' argument seems not to work right.  Maybe you meant to use '--' or '--name-status'?

**Comment by franrivasortega** [comment]
> Everything is very nice, except the 'file' argument seems not to work right. Maybe you meant to use '--' or '--name-status'?

You're right, I overthinked and used `follow`, which is an option used for other purposes. `--` is the right option, I'll fix it so the user can choose multiple files through a character vector.

**Comment by matloff** [comment]
Thanks, merged.

**Pull request merged**